QualityControl/lib/middleware/layouts/layoutId.middleware.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { InvalidInputError } from '../../errors/InvalidInputError.js';
+import { updateExpressResponseFromNativeError } from '../../errors/updateExpressResponseFromNativeError.js';
+
+/**
+ * Middleware that checks if the layout id is present in the request
+ * @param {JSONFileConnector} dataService - service for getting/setting layout data
+ * @returns  {function(req, res, next): Function} - middleware function
+ */
+export const layoutIdMiddleware = (dataService) =>
+
+/**
+ * Returned middleware method
+ * @param {Express.Request} req - HTTP Request
+ * @param {Express.Response} res - HTTP Response
+ * @param {Express.Next} next - HTTP Next (check pass)
+ */
+  async (req, res, next) => {
+    const { id = '' } = req.params ?? {};
+    try {
+      if (!dataService) {
```

Comment:
Dataservice is not input from the client request but instead a service from the server. Thus, a mroe suitable error would be with status code 503 - service unavailable. 

But I would highly recommend you extract this in a different middleware so that the function usecase is kept simple and related to its definition